src/server/scripts/Northrend/Ulduar/HallsOfLightning/instance_halls_of_lightning.cpp

```diff
@@ -60,6 +63,27 @@ class instance_halls_of_lightning : public InstanceMapScript
                 LoadDoorData(doorData);
             }
 
+            void OnCreatureCreate(Creature* creature) override
+            {
+                InstanceScript::OnCreatureCreate(creature);
+
+                switch (creature->GetEntry())
+                {
+                    case NPC_MOLTEN_GOLEM:
+                        if (GetBossState(DATA_VOLKHAN) == IN_PROGRESS)
+                        {
+                            if (Creature* volkhan = GetCreature(DATA_VOLKHAN))
+                                if (CreatureAI* ai = volkhan->AI())
+                                    ai->JustSummoned(creature);
+                        }
+                        else // These golems are summoned via trigger missile so we have to clean them up if they spawned during a wipe/completion
+                            creature->DespawnOrUnsummon();
```

Comment:
#28102 if you're interested

Reply:
I'm not sure if this is actually related to your issue because you refer to direct summons, while this one here is summoned by trigger missies which are timed correctly.